ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java

```diff
@@ -457,14 +452,15 @@ public Builder valueColumns(final Iterable<? extends Column> column) {
     }
 
     public LogicalSchema build() {
+      final List<Column> metadata = metadataBuilder.build();
       final List<Column> suppliedKey = keyBuilder.build();
 
       final List<Column> key = suppliedKey.isEmpty()
           ? IMPLICIT_KEY_SCHEMA
           : suppliedKey;
 
       return new LogicalSchema(
-          METADATA_SCHEMA,
+          metadata.isEmpty() ? METADATA_SCHEMA : metadata,
```

Comment:
Not entirely understanding this change. Is the idea that external Builder instances will continue to use the default metadata schema, but that internal Builder instances will copy the metadata around? 

The metadata schema is not updated anywhere, is it? So this is equivalent to always returning the default metadata schema, except "safer" in that we copy when we should. Is that correct?

Reply:
I thought the comment would help here - basically the metadata is actually updated when we add/remove aliases :( I'll be trying to remove that in the long run.

Reply:
Ah got'cha. And all of that is still managed by this class, which is why the `metadataColumns()` method can be private. Thanks for clarifying!